apphub/semantic_segmentation/unet_montgomery/unet_montgomery_tf.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2019 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""U-Net example."""
+import tempfile
+from typing import Any, Dict, List
+
+import cv2
+import numpy as np
+import tensorflow as tf
+
+import fastestimator as fe
+from fastestimator.architecture.tensorflow import UNet
+from fastestimator.dataset import montgomery
+from fastestimator.op import NumpyOp
+from fastestimator.op.numpyop import Delete, HorizontalFlip, Minmax, ReadImage, Resize, Rotate, Sometimes
+from fastestimator.op.tensorop.loss import CrossEntropy
+from fastestimator.op.tensorop.model import ModelOp, UpdateOp
+from fastestimator.trace.io import BestModelSaver
+from fastestimator.trace.metric import Dice
+
+
+class CombineLeftRightMask(NumpyOp):
+    """NumpyOp to combine left lung mask and right lung mask."""
+    def forward(self, data: List[np.ndarray], state: Dict[str, Any]) -> List[np.ndarray]:
+        mask_left, mask_right = data
+        data = mask_left + mask_right
+        return data
+
+
+def get_estimator(epochs=20, batch_size=4, max_steps_per_epoch=None, save_dir=tempfile.mkdtemp(), log_steps=20):
+    # step 1
+    csv = montgomery.load_data()
+    pipeline = fe.Pipeline(
+        train_data=csv,
+        eval_data=csv.split(0.2),
+        batch_size=batch_size,
+        ops=[
+            ReadImage(inputs="image", parent_path=csv.parent_path, outputs="image", grey_scale=True),
+            ReadImage(inputs="mask_left",
+                      parent_path=csv.parent_path,
+                      outputs="mask_left",
+                      grey_scale=True,
+                      mode='!infer'),
```

Comment:
why would you not read image during inference?

Reply:
During inferencing, I don't have masks. Only image is sent to the network.

Reply:
oh right